sway/desktop/xwayland.c

```diff
@@ -77,6 +77,18 @@ static void unmanaged_handle_unmap(struct wl_listener *listener, void *data) {
 	desktop_damage_surface(xsurface->surface, xsurface->x, xsurface->y, true);
 	wl_list_remove(&surface->link);
 	wl_list_remove(&surface->commit.link);
+
+	if (!wlr_xwayland_surface_is_unmanaged(xsurface)) {
```

Comment:
Not sure about what unmanaged means in the first place so I might be off the mark, but for layer shell you check if the seat's focus is on the unmapped surface ; this might need a similar check here?
You wouldn't want focus to change because some x11 window depops in another workspace due to a timeout or something.

Reply:
I've added a similar check. To test, I launched a terminal and Firefox side by side. In the terminal I ran `i3-input & sleep 5 && killall i3-input`, then while the sleep was occurring I opened the settings menu of Firefox and waited. After unmapping, the Firefox menu remained open and had keyboard focus.